lib/sinon/match.js

```diff
@@ -221,12 +221,50 @@ match.hasOwn = createPropertyMatcher(function (actual, property) {
     return actual.hasOwnProperty(property);
 }, "hasOwn");
 
+match.array = match.typeOf("array");
+
+match.array.deepEquals = function (expectation) {
+    return match(function (actual) {
+        // Comparing lengths is the fastest way to spot a difference before iterating through every item
+        var sameLength = actual.length === expectation.length;
+        return sameLength && Array.prototype.every.call(actual, function (element, index) {
```

Comment:
However, I was just thinking if we should act upon those iterables as if they were arrays, after all this marcher has the word `array` in it.

Please let me know what you think, I can change this without problems if you want me to :smile:

Also, thank you both for the feedback, it's always good to have such a careful review. I usually learn a lot with it.